manual/vcredist2008/tools/chocolateyUninstall.ps1

```diff
@@ -1,15 +1,28 @@
-﻿$packageName = 'vcredist2005'
-$packageSearch = "Microsoft Visual C++ 2008 Redistributable*"
-$installerType = 'msi'
-$silentArgs = '/quiet /qn /norestart'
-$validExitCodes = @(0,3010)  # http://msdn.microsoft.com/en-us/library/aa368542(VS.85).aspx
+﻿$ErrorActionPreference = 'Stop'
 
-Get-ItemProperty -Path @( 'HKLM:\Software\Wow6432Node\Microsoft\Windows\CurrentVersion\Uninstall\*',
-                          'HKLM:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*',
-                          'HKCU:\Software\Microsoft\Windows\CurrentVersion\Uninstall\*' ) `
-                 -ErrorAction:SilentlyContinue `
-| Where-Object   {$_.DisplayName -Like $packageSearch} `
-| ForEach-Object {Uninstall-ChocolateyPackage -PackageName "$packageName" `
-                                              -FileType "$installerType" `
-                                              -SilentArgs "$($_.PSChildName) $silentArgs" `
-                                              -ValidExitCodes $validExitCodes}
\ No newline at end of file
+$packageName         = 'vcredist2008'
+$softwareNamePattern = 'Microsoft Visual C++ 2008 Redistributable*'
+
+[array] $key = Get-UninstallRegistryKey $softwareNamePattern
+if ($key.Count -eq 1) {
+    $key | % {
+        $packageArgs = @{
+            packageName            = $packageName
+            silentArgs             = "/quiet /qn /norestart"
+            fileType               = 'msi'
+            validExitCodes         = @(0,3010) # http://msdn.microsoft.com/en-us/library/aa368542(VS.85).aspx
+            file                   = ''
+        }
+        $packageArgs.file = "$($_.UninstallString.Replace(' /x86=0', ''))"   #"C:\Program Files\OpenSSH\uninstall.exe" /x86=0
+        Uninstall-ChocolateyPackage @packageArgs
+    }
+}
+elseif ($key.Count -eq 0) {
+    Write-Warning "$packageName has already been uninstalled by other means."
+}
+elseif ($key.Count -gt 1) {
+    Write-Warning "$key.Count matches found!"
```

Comment:
this won't work correctly I believe.
Wouldn't this output the value of `$key` and then append `.Count` instead?